From 670e314c2737300ea618d7e3f51779af9e5009d2 Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Tue, 2 Apr 2024 11:56:09 +0200
Subject: [PATCH 08/14] FIX: wallet listener crashing

---
 src/wallet/api/wallet.cpp | 21 +++++++++++++++------
 1 file changed, 15 insertions(+), 6 deletions(-)

diff --git a/src/wallet/api/wallet.cpp b/src/wallet/api/wallet.cpp
index ed309de34..ee0eeeb01 100644
--- a/src/wallet/api/wallet.cpp
+++ b/src/wallet/api/wallet.cpp
@@ -201,8 +201,11 @@ struct Wallet2CallbackImpl : public tools::i_wallet2_callback
                      << ", burnt: " << print_money(burnt)
                      << ", raw_output_value: " << print_money(amount)
                      << ", idx: " << subaddr_index);
-        m_listener->moneyReceived(tx_hash, amount);
-        m_listener->updated();
+        // do not signal on sent tx if wallet is not syncronized completely
+        if (m_listener && m_wallet->synchronized()) {
+            m_listener->moneyReceived(tx_hash, amount);
+            m_listener->updated();
+        }
     }
 
     virtual void on_unconfirmed_money_received(uint64_t height, const crypto::hash &txid, const cryptonote::transaction& tx, uint64_t amount, const cryptonote::subaddress_index& subaddr_index)
@@ -214,8 +217,11 @@ struct Wallet2CallbackImpl : public tools::i_wallet2_callback
                      << ", tx: " << tx_hash
                      << ", amount: " << print_money(amount)
                      << ", idx: " << subaddr_index);
-        m_listener->unconfirmedMoneyReceived(tx_hash, amount);
-        m_listener->updated();
+        // do not signal on sent tx if wallet is not syncronized completely
+        if (m_listener && m_wallet->synchronized()) {
+            m_listener->unconfirmedMoneyReceived(tx_hash, amount);
+            m_listener->updated();
+        }
     }
 
     virtual void on_money_spent(uint64_t height, const crypto::hash &txid, const cryptonote::transaction& in_tx,
@@ -227,8 +233,11 @@ struct Wallet2CallbackImpl : public tools::i_wallet2_callback
                      << ", tx: " << tx_hash
                      << ", amount: " << print_money(amount)
                      << ", idx: " << subaddr_index);
-        m_listener->moneySpent(tx_hash, amount);
-        m_listener->updated();
+        // do not signal on sent tx if wallet is not syncronized completely
+        if (m_listener && m_wallet->synchronized()) {
+            m_listener->moneySpent(tx_hash, amount);
+            m_listener->updated();
+        }
     }
 
     virtual void on_skip_transaction(uint64_t height, const crypto::hash &txid, const cryptonote::transaction& tx)
-- 
2.39.2

