From ecdae80357168d532bc7625f337c6dff8b24939e Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Tue, 2 Apr 2024 17:07:19 +0200
Subject: [PATCH 09/14] fix missing ___clear_cache when targetting iOS

---
 .gitmodules | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/.gitmodules b/.gitmodules
index bd30f0444..fa54f48b1 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -20,6 +20,6 @@
 	branch = monero
 [submodule "external/randomwow"]
 	path = external/randomwow
-	url = https://github.com/wownero-mirror/RandomWOW
-        branch = 1.2.1-wow
+	url = https://github.com/mrcyjanek/RandomWOW
+        branch = cyjan-fix-ios
 
-- 
2.39.2

