From 8cfcf5b132c41aedbc9c9a0932ffb832cf410761 Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Tue, 2 Apr 2024 17:07:19 +0200
Subject: [PATCH 9/9] fix missing ___clear_cache when targetting iOS

---
 .gitmodules        | 4 ++--
 external/randomwow | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/.gitmodules b/.gitmodules
index 86009b6b6..f723bd2d8 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -20,6 +20,6 @@
 	url = https://github.com/tevador/polyseed.git
 [submodule "external/randomwow"]
 	path = external/randomwow
-	url = https://git.wownero.com/wownero/RandomWOW
-        branch = 1.2.1-wow
+	url = https://github.com/mrcyjanek/RandomWOW
+        branch = cyjan-fix-ios
 
diff --git a/external/randomwow b/external/randomwow
index 27b099b6d..6f30d4b92 160000
--- a/external/randomwow
+++ b/external/randomwow
@@ -1 +1 @@
-Subproject commit 27b099b6dd6fef6e17f58c6dfe00009e9c5df587
+Subproject commit 6f30d4b924fecb231e5b683915cc75d18b3b5866
-- 
2.44.0

