From aabafcfc0572651436d024a635483c49042fad7f Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Thu, 28 Mar 2024 00:32:51 +0100
Subject: [PATCH] disable soname

---
 CMakeLists.txt | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a8e7c2..5301353 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,6 +36,7 @@ include_directories(polyseed
 target_compile_definitions(polyseed PRIVATE POLYSEED_SHARED)
 set_target_properties(polyseed PROPERTIES VERSION 2.0.0
                                           SOVERSION 2
+                                          NO_SONAME 1
                                           C_STANDARD 11
                                           C_STANDARD_REQUIRED ON)

@@ -45,16 +46,17 @@ include_directories(polyseed_static
   include/)
 target_compile_definitions(polyseed_static PRIVATE POLYSEED_STATIC)
 set_target_properties(polyseed_static PROPERTIES OUTPUT_NAME polyseed
+                                                 NO_SONAME 1
                                                  C_STANDARD 11
                                                  C_STANDARD_REQUIRED ON)

-add_executable(polyseed-tests
-  tests/tests.c)
-include_directories(polyseed-tests
-  include/)
-target_compile_definitions(polyseed-tests PRIVATE POLYSEED_STATIC)
-target_link_libraries(polyseed-tests
-  PRIVATE polyseed_static)
+# add_executable(polyseed-tests
+#   tests/tests.c)
+# include_directories(polyseed-tests
+#   include/)
+# target_compile_definitions(polyseed-tests PRIVATE POLYSEED_STATIC)
+# target_link_libraries(polyseed-tests
+#   PRIVATE polyseed_static)

 include(GNUInstallDirs)
 install(TARGETS polyseed polyseed_static
--
2.39.2
