From 186a6a89cd6d00e4269408062ba64be60a6a80c9 Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Tue, 2 Apr 2024 16:51:56 +0200
Subject: [PATCH 01/21] fix missing ___clear_cache when targetting iOS

---
 .gitmodules      | 3 ++-
 external/randomx | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/.gitmodules b/.gitmodules
index 31c6921e5..1c28e307e 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -6,7 +6,8 @@
 	url = https://github.com/trezor/trezor-common.git
 [submodule "external/randomx"]
 	path = external/randomx
-	url = https://github.com/tevador/RandomX
+	url = https://github.com/MrCyjaneK/RandomX
+	branch = cyjan-fix-ios-v2
 [submodule "external/supercop"]
 	path = external/supercop
 	url = https://github.com/monero-project/supercop
diff --git a/external/randomx b/external/randomx
index 12f2c2ffe..ab495666c 160000
--- a/external/randomx
+++ b/external/randomx
@@ -1 +1 @@
-Subproject commit 12f2c2ffe2108d6cf54c391fee33c8bc3646cdab
+Subproject commit ab495666ca1f77800bf90e6f16ecc8307abf0092
-- 
2.55.0

