From ac747342eb270cf09e6a09b41b89548509d5882b Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Tue, 2 Apr 2024 16:51:56 +0200
Subject: [PATCH] fix missing ___clear_cache when targetting iOS

---
 .gitmodules      | 2 +-
 external/randomx | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/.gitmodules b/.gitmodules
index 73a23fb35..7ea87a009 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -9,7 +9,7 @@
 	url = https://github.com/trezor/trezor-common.git
 [submodule "external/randomx"]
 	path = external/randomx
-	url = https://github.com/tevador/RandomX
+	url = https://github.com/MrCyjaneK/RandomX
 [submodule "external/utf8proc"]
 	path = external/utf8proc
 	url = https://github.com/JuliaStrings/utf8proc.git
diff --git a/external/randomx b/external/randomx
index 102f8acf9..cccda3a50 160000
--- a/external/randomx
+++ b/external/randomx
@@ -1 +1 @@
-Subproject commit 102f8acf90a7649ada410de5499a7ec62e49e1da
+Subproject commit cccda3a50f8f5aeb1a7520901d822085fd7881c5
-- 
2.44.0

