From 3e83b0212fe9a824ff9e8c399ef86e25c73afa13 Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Mon, 18 Nov 2024 10:57:37 -0500
Subject: [PATCH 13/22] include locale only when targeting WIN32

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 133997f74..95326743c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -953,7 +953,8 @@ elseif(Boost_FOUND)
   message(STATUS "Found Boost Version: ${Boost_VERSION_STRING}")
 
   set(BOOST_COMPONENTS filesystem thread date_time chrono serialization program_options)
-  if (WIN32)
+
+  if(WIN32)
     list(APPEND BOOST_COMPONENTS locale)
   endif()
 
-- 
2.54.0 (Apple Git-157)

