--- a/src/llvm-project/compiler-rt/lib/builtins/assembly.h	2025-07-10 12:41:44.469384349 +0200
+++ b/src/llvm-project/compiler-rt/lib/builtins/assembly.h	2025-07-10 12:46:03.694440678 +0200
@@ -254,9 +254,10 @@
   .globl name SEPARATOR                                                        \
   SYMBOL_IS_FUNC(name) SEPARATOR                                               \
   DECLARE_SYMBOL_VISIBILITY_UNMANGLED(name) SEPARATOR                          \
+  name: SEPARATOR                                                              \
   CFI_START SEPARATOR                                                          \
   DECLARE_FUNC_ENCODING                                                        \
-  name: SEPARATOR BTI_C
+  BTI_C
 
 #define DEFINE_COMPILERRT_FUNCTION_ALIAS(name, target)                         \
   .globl SYMBOL_NAME(name) SEPARATOR                                           \
