From 865ffb66cbaeff20115b812f1380bb15c9844129 Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Fri, 1 Nov 2024 17:29:35 +0100
Subject: [PATCH] boost fix filenames

---
 contrib/depends/packages/boost.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/depends/packages/boost.mk b/contrib/depends/packages/boost.mk
index ff725680..be382d3a 100644
--- a/contrib/depends/packages/boost.mk
+++ b/contrib/depends/packages/boost.mk
@@ -9,7 +9,7 @@ $(package)_patches=fix_aroptions.patch fix_arm_arch.patch fix_pthread_win.patch
 define $(package)_set_vars
 $(package)_config_opts_release=variant=release
 $(package)_config_opts_debug=variant=debug
-$(package)_config_opts=--layout=tagged --build-type=complete --user-config=user-config.jam
+$(package)_config_opts=--layout=system --user-config=user-config.jam
 $(package)_config_opts+=threading=multi link=static -sNO_BZIP2=1 -sNO_ZLIB=1
 $(package)_config_opts_linux=threadapi=pthread runtime-link=static
 $(package)_config_opts_android=threadapi=pthread runtime-link=static target-os=android
-- 
2.39.5 (Apple Git-154)

