From 2ca9fcd1b79c687ab95495b84deac0eed58c81ef Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Sat, 7 Mar 2026 00:20:47 +0100
Subject: [PATCH 10/11] workaround: no-op MARK_AS_POD_C11

---
 contrib/epee/include/misc_language.h | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/contrib/epee/include/misc_language.h b/contrib/epee/include/misc_language.h
index 04cb9750..7d512c3d 100644
--- a/contrib/epee/include/misc_language.h
+++ b/contrib/epee/include/misc_language.h
@@ -39,16 +39,7 @@
 #include "auto_val_init.h"
 
 
-#define MARK_AS_POD_C11(type)   \
-namespace std  \
-{  \
-  template<> \
-struct is_pod< type > \
-  { \
-  static const bool value = true; \
-  }; \
-}
-
+#define MARK_AS_POD_C11(type)
 
 namespace epee
 {
-- 
2.54.0 (Apple Git-156)

